H5Ztrans: (feature) Improved H5Z_xform_noop() and H5Z_xform_create() …

jwsblokland opened the pull request:
Improved H5Z_xform_noop() and H5Z_xform_create() function

- Made a small improvement for H5Z_xform_noop() function. Now,
  the function will also return TRUE if the data transform function
  expression = "x". For this case, the HDF5 library behaves in a
  similar fashion as the case when no data transform function has been
  specified.
- Improved the inline documentation of the function
  H5Z_xform_create() such it is more inline with the rest of the
  code.

This pull request has been requested in the abandoned pull request https://github.com/HDFGroup/hdf5/pull/922.

Commit:
H5Ztrans: (feature) Improved H5Z_xform_noop() and H5Z_xform_create() function

- Made a small improvement for H5Z_xform_noop() function. Now,
  the function will also return TRUE if the data transform function
  expression = "x". For this case, the HDF5 library behaves in a
  similar fashion as the case when no data transform function has been
  specified.
- Improved the inline documentation of the function
  H5Z_xform_create() such it is more inline with the rest of the
  code.

github-actions[bot] committed:
Committing clang-format changes

qkoziol reviewed (changes requested):
This code looks fine, but it needs a test, to demonstrate that it actually works.  Adding it to both the serial testing for data transforms (in the 'test' directory) and the parallel testing (in the 'testpar' directory) is probably necessary, since I believe that use in parallel HDF5 is desired.

qkoziol commented:
> This code looks fine, but it needs a test, to demonstrate that it actually works. Adding it to both the serial testing for data transforms (in the 'test' directory) and the parallel testing (in the 'testpar' directory) is probably necessary, since I believe that use in parallel HDF5 is desired.

Also, with a test, you'll guarantee that it stays working. :-)

Commit:
H5Ztrans: (feature) Added 3 tests for improved H5Z_xform_noop() function

- Added serial test with data transform expression = "x" to
  verify the improved H5Z_xform_noop() function behaves as expected.
- Added 2 parallel tests with data transform expression = "x"
  in combination with a filter. Before, these tests will fail but
  with the improved H5Z_xform_noop() function they work and result
  in the expected behavior.
- Small bug fix for one of parallel filter tests.

github-actions[bot] committed:
Committing clang-format changes

jwsblokland commented:
As requested I have added 1 serial test and 2 parallel tests. I must admit usually I add some tests to demostrates the feature or bug fix works as expected. Apparently, this case I did not, so thank you for reminding me to add these tests.

epourmal commented on testpar/t_filters_parallel.c:
```diff
@@ -1287,6 +1291,177 @@ test_write_filtered_dataset_interleaved_write(void)
+ * the transform function. Therefore it does not apply
+ * the transform function resulting in not breaking to
+ * independent I/O.
+ *
```
Please add the comment to the RELEASE.txt file (similar to what you have here) explaining the change. It should go under Bug Fixed since HDF5 ... / Library.
Thank you!

jwsblokland replied:
@epourmal , I have added a comment about the implemented improvement to the `RELEASE.txt` file. Thanks for mentioning this.

epourmal replied:
Thank you! We are currently testing on HPC systems now and will accept the change.

Commit:
H5Ztrans: (feature) Added release note about detection of the
           simple data transform function "x".

- Added a brief explanation about the implemented improvement
  of the detection of the simple data transform function "x"
  to the RELEASE.txt file.

lrknox commented:
The tests added also passed on jelly, lassen quartz and mutrino.
